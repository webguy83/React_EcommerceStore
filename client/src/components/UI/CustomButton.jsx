import React from 'react';
import styled, { css } from 'styled-components';

// css
const GoogleSignIn = css`
    background-color: #4285F4;
    color: white;
    border: none;
    
    &:hover {
        background-color: #f4b400
    }
`

const AddToCartBtn = css`
    background-color: #f4b400
    color: black;

    &:hover {
        background-color: white;
    }
`

const CustomButtonContainer = styled.button`
    min-width: 11rem;
    padding: 1rem;
    cursor: pointer;
    text-transform: uppercase;
    font-family: inherit;
    font-size: 1.4rem;
    font-weight: bold;
    border: .1rem solid white;
    background-color: black;
    color: white;
    outline: none;
    transition: all .3s ease-out;

    &:hover, &:active {
        color: black;
        background-color: white;
        transition: all .3s ease-out;
    }

    ${({ defaultColour }) => {
        return defaultColour;
    }}
`
// jsx

const CustomButton = ({ type, value, click, google, addToCart, ...otherProps}) => {
    const defaultColour = google ? GoogleSignIn : addToCart ? AddToCartBtn : "";
    return (
        <CustomButtonContainer onClick={click}
            type={type}
            defaultColour={defaultColour}
            {...otherProps}
            >{value}
        </CustomButtonContainer>
    );
};

export default CustomButton;