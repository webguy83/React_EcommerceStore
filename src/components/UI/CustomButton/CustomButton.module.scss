.customButtonContainer {
    min-width: 11rem;
    padding: 1rem;
    cursor: pointer;
    text-transform: uppercase;
    font-family: inherit;
    font-size: 1.4rem;
    font-weight: bold;
    border: .1rem solid black;
    background-color: black;
    color: white;
    outline: none;
    transition: all .2s ease-out;

    &:hover, &:active {
        color: black;
        background-color: white;
        transition: all .2s ease-out;
    }

    &.googleSignIn {
        background-color: #4285F4;
        color: white;
        border: none;
    
        &:hover {
            background-color: #F4B400;
        }
    }

    &.addToCartBtn {
        background-color: #F4B400;
        color: black;

        &:hover {
            background-color: white;
        }
    }
}