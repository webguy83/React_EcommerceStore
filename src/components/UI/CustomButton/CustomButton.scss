.customButton {
    width: 40%;
    min-width: 11rem;
    padding: 1rem;
    cursor: pointer;
    text-transform: uppercase;
    font-family: inherit;
    font-size: 1.4rem;
    border: .1rem solid black;
    background-color: black;
    color: white;
    outline: none;

    &:hover, &:active {
        color: black;
        background-color: white;
    }
}
