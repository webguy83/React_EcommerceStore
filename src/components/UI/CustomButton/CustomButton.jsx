import React from 'react';

import './CustomButton.scss';

export default ({type, value}) => {
    return (
        <button type={type} className="customButton">{value}</button>
    );
};