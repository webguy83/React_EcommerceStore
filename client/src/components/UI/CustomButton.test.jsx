import React from "react";
import CustomButton from "./CustomButton";
import "jest-styled-components";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("default tests for CustomButton", () => {
  afterEach(cleanup);

  it("should render without errors", () => {
    const { asFragment } = render(<CustomButton />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with a value specified", () => {
    const { getByTestId } = render(<CustomButton value="MyButton" />);
    expect(getByTestId("container")).toHaveTextContent(/^MyButton$/);
  });

  it("should be a submit button type when specified", () => {
    const { getByTestId } = render(<CustomButton type="submit" />);
    expect(getByTestId("container").type).toBe("submit");
  });

  it("should show a google colour for google sign in", () => {
    const { getByTestId } = render(<CustomButton google={true} />);
    expect(getByTestId("container")).toHaveStyleRule(
      "background-color",
      "#4285f4"
    );
  });

  it("should show a default colour of black for the button", () => {
    const { getByTestId } = render(<CustomButton google={false} />);
    expect(getByTestId("container")).toHaveStyleRule(
      "background-color",
      "var(--prim)"
    );
  });
});
