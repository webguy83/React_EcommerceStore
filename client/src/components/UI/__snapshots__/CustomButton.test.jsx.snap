// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default tests for CustomButton should render without errors 1`] = `
<DocumentFragment>
  .c0 {
  min-width: 11rem;
  padding: 1rem;
  cursor: pointer;
  text-transform: uppercase;
  font-family: inherit;
  font-size: 1.4rem;
  font-weight: bold;
  border: 0.1rem solid white;
  background-color: var(--prim);
  color: white;
  outline: none;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

.c0:hover,
.c0:active {
  background-color: var(--sec);
  -webkit-transition: var(--btnTrans);
  transition: var(--btnTrans);
}

<button
    class="c0"
    data-testid="container"
    type="button"
  />
</DocumentFragment>
`;
